### ISSUE TITLE
Sending code seems to give several errors in data. Recieve working perfectly.

### ISSUE THREAD

**Issue by pibrochman** [latonita/arduino-canbus-monitor#2]
Hi I built you code onto an arduino UNO, with CAN module with 16MHz clock. When I used the CANHACKER software to send messages they do not get sent correctly. I put a loop back onto another comm port and the comport serial messages seem to be correctly formatted as per the Lawicel spec. If a message is sent with DLC of 8, no data bytes are sent. If the DLC is less than 8, the bytes are sent but are wrong.  If I send a can message with the following: ID006, DLC7, 12,34,56,78,50,40,30, what is recieved is ID006, DLC7, 22,44,66,88,00,00,00. Any ideas?   Hardware setup: I checked this with a Peakcan USB, and IXXAT tincan, both showed same messages. By the way great project, and a real good way for people to get a low cost can analyser. Hope this info helps. Regards

**Comment by latonita** [comment]
Hi,
- did you tried the latest version of arduino-canbus-monitor? i incorporated few changes couple days ago
- have you tried examples from MCP library to send data without my code?
- which chips are on the board? mcp2515+mcp2551 or mcp2515+tjaXXXX ?
- do you send regular or extended frame? i suppose regular
- which speed?

recently I found and logged quite similar error with 16mhz module. 8mhz was working fine. looking into this, but it might take time due to my availability

**Comment by pibrochman** [comment]
Hi Anton

I tried your recently updated code and its now working well.

I'm using a ch340 usb-ttl (£1)+ Aurduino pro mini (£1)+ Arduino MCP2515 
CAN Bus Module TJA1050 Receiver SPI Module (£1.50). I had previously 
changed the crystal on the CAN module to 16MHz.

I have now tried the code using the original 8MHz crystal, having 
changed the code call to 8MHzand its all working well(I used an arduino 
nano instead £1.50)

So for £3.00-3.50 that makes a great little CAN analyser. What an 
excellent project, well done and thank you. The CANhacker software is 
also very good.

Kind Regards
Matthew
